numba/tests/test_fancy_indexing.py

```diff
@@ -176,6 +176,52 @@ def test_setitem_array(self):
         indices = self.generate_advanced_indices(N)
         self.check_setitem_indices(arr, indices)
 
+    def test_setitem_0d(self):
+        # Test setitem with a 0d-array
+        pyfunc = setitem_usecase
+        cfunc = jit(nopython=True)(pyfunc)
+
+        x1 = np.zeros(3)
+        v1 = np.array(3.14)
+        exp1 = pyfunc(x1.copy(), 0, v1)
+        got1 = cfunc(x1.copy(), 0, v1)
+        self.assertPreciseEqual(exp1, got1)
+
+        exp2 = pyfunc(x1.copy(), 2, v1)
+        got2 = cfunc(x1.copy(), 2, v1)
+        self.assertPreciseEqual(exp2, got2)
+
+        # --------------
+        x2 = np.zeros(3, dtype=np.int64)
+        v2 = np.array(3, dtype=np.int64)
+        exp3 = pyfunc(x2.copy(), 0, v2)
+        got3 = cfunc(x2.copy(), 0, v2)
+        self.assertPreciseEqual(exp3, got3)
+
+        # --------------
+        # coercing
+        x_f64 = np.zeros(3, dtype=np.float64)
+        v_i64 = np.array(1, dtype=np.int64)
+        exp4 = pyfunc(x_f64.copy(), 0, v_i64)
+        got4 = cfunc(x_f64.copy(), 0, v_i64)
+        self.assertPreciseEqual(exp4, got4)
+
+        # --------------
+        # UnicodeCharSeq
+        x_ucs = np.zeros(5, dtype='<U3')
+        v_ucs = np.array('abc')
+        exp5 = pyfunc(x_ucs.copy(), 0, v_ucs)
+        got5 = cfunc(x_ucs.copy(), 0, v_ucs)
+        self.assertPreciseEqual(exp5, got5)
+
+        # --------------
+        # Complex
+        x_cp = np.zeros(3, dtype=np.complex)
+        v_cp = np.array(2+3j, dtype=np.complex)
+        exp5 = pyfunc(x_cp.copy(), 0, v_cp)
+        got5 = cfunc(x_cp.copy(), 0, v_cp)
+        self.assertPreciseEqual(exp5, got5)
```

Comment:
I wonder if the 0-d arrays can be the induction variables of a loop and all the variations can be tested with the same code block calling `setitem_usecase`?